### PR TITLE
feature(secretstores): add context to api and tencent/ssm

### PR DESCRIPTION
**Pull request by 1046102779**
https://github.com/dapr/components-contrib/pull/1973
/cc @berndverst @ItalyPaleAle

### CODE REVIEW

File: pkg/encryption/encryption.go [hunk 1]
```diff
@@ -127,7 +128,7 @@ func tryGetEncryptionKeyFromMetadataItem(namespace string, item v1alpha1.Metadat
 		return Key{}, errors.Errorf("%s: secretKeyRef cannot be empty", errPrefix)
 	}
 
-	r, err := secretStore.GetSecret(secretstores.GetSecretRequest{
+	r, err := secretStore.GetSecret(context.Background(), secretstores.GetSecretRequest{
```
Comment (ItalyPaleAle) [inline pkg/encryption/encryption.go:131]: Please use `context.TODO()` here and/or add a comment `TODO: cascade context`

File: pkg/runtime/runtime.go [hunk 2]
```diff
@@ -2397,7 +2397,7 @@ func (a *DaprRuntime) processComponentSecrets(component components_v1alpha1.Comp
 
 		resp, ok := cache[m.SecretKeyRef.Name]
 		if !ok {
-			r, err := secretStore.GetSecret(secretstores.GetSecretRequest{
+			r, err := secretStore.GetSecret(context.Background(), secretstores.GetSecretRequest{
```
Comment (ItalyPaleAle) [inline pkg/runtime/runtime.go:2400]: Here too. Please use `context.TODO()` and/or add a comment: `TODO: Cascade context`

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: pkg/runtime/runtime.go (hunk 2 above)
Comment (1046102779) [reply pkg/runtime/runtime.go]: Fixed.

File: pkg/encryption/encryption.go (hunk 1 above)
Comment (1046102779) [reply pkg/encryption/encryption.go]: Fixed.

**Comment by 1046102779** [comment]
This PR needs to wait for [another MR](https://github.com/dapr/components-contrib/pull/1973)  to be merged,  before it can be changed back in the `go.mod` file, a temporarily used fork repo `github.com/1046102779/components-contrib`

**Comment by yaron2** [comment]
@1046102779 we also need a docs PR for this component.

**Comment by 1046102779** [comment]
> @1046102779 we also need a docs PR for this component.

docs PR: https://github.com/dapr/docs/pull/2732

**Review by daixiang0 — DISMISSED** [review]

**Comment by 1046102779** [comment]
> This PR needs to wait for [another PR](https://github.com/dapr/components-contrib/pull/1973) to be merged, before it can be changed back in the `go.mod` file, a temporarily used fork repo `github.com/1046102779/components-contrib`

Done. /cc @yaron2 @berndverst @daixiang0 @ItalyPaleAle @artursouza

**Comment by 1046102779** [comment]
fixed: 
>  make lint 
```shell
pkg/components/state/pluggable_test.go:617:5  typecheck  unknown field `Filters` in struct literal
```

image

File: pkg/encryption/encryption.go [hunk 3]
```diff
@@ -127,7 +128,8 @@ func tryGetEncryptionKeyFromMetadataItem(namespace string, item v1alpha1.Metadat
 		return Key{}, errors.Errorf("%s: secretKeyRef cannot be empty", errPrefix)
 	}
 
-	r, err := secretStore.GetSecret(secretstores.GetSecretRequest{
+	// TODO: cascade context.
+	r, err := secretStore.GetSecret(context.TODO(), secretstores.GetSecretRequest{
```
Comment (berndverst) [inline pkg/encryption/encryption.go:132]: why not `context.Background()` ?

File: pkg/runtime/runtime.go [hunk 4]
```diff
@@ -2494,7 +2494,8 @@ func (a *DaprRuntime) processComponentSecrets(component componentsV1alpha1.Compo
 
 		resp, ok := cache[m.SecretKeyRef.Name]
 		if !ok {
-			r, err := secretStore.GetSecret(secretstores.GetSecretRequest{
+			// TODO: cascade context.
+			r, err := secretStore.GetSecret(context.TODO(), secretstores.GetSecretRequest{
```
Comment (berndverst) [inline pkg/runtime/runtime.go:2498]: Why not `context.Background()` ?

File: pkg/components/state/pluggable_test.go [hunk 5]
```diff
@@ -650,7 +650,9 @@ func TestComponentCalls(t *testing.T) {
 		}
 		request := &state.QueryRequest{
 			Query: query.Query{
-				Filters: filters,
+				QueryFields: query.QueryFields{
+					Filters: filters,
+				},
```
Comment (halspang) [inline pkg/components/state/pluggable_test.go:655]: What's this doing here?

File: pkg/encryption/encryption.go (hunk 3 above)
Comment (berndverst) [reply pkg/encryption/encryption.go:132]:
```suggestion
	r, err := secretStore.GetSecret(context.Background(), secretstores.GetSecretRequest{
```

File: pkg/runtime/runtime.go (hunk 4 above)
Comment (berndverst) [reply pkg/runtime/runtime.go:2506]:
```suggestion
			r, err := secretStore.GetSecret(context.Background(), secretstores.GetSecretRequest{
```

Comment (ItalyPaleAle) [reply pkg/runtime/runtime.go:2506]: `context.TODO` iscorrect here as it's a placeholder for cascading context in the future

**Review by ItalyPaleAle — APPROVED** [review]

**Review by berndverst — APPROVED** [review]

**Comment by berndverst** [comment]
@halspang - the pluggable components change is due to https://github.com/dapr/components-contrib/pull/2038

File: pkg/components/state/pluggable_test.go (hunk 5 above)
Comment (halspang) [reply pkg/components/state/pluggable_test.go:655]: @berndverst explained why this is here, nevermind.

**Review by halspang — APPROVED** [review]

File: pkg/components/state/pluggable_test.go (hunk 5 above)
Comment (berndverst) [reply pkg/components/state/pluggable_test.go:655]: The Query interface was changed in contrib.

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**